ReactAndroid/src/main/java/com/facebook/react/devsupport/BundleDownloader.java

```diff
@@ -98,91 +102,105 @@ public int getFilesChangedCount() {
     }
   }
 
+  private interface DownloadListener {
+    void onSuccess(String sourceURL, @Nullable NativeDeltaClient nativeDeltaClient, @Nullable String[] additionalBundles);
+    void onProgress(@Nullable String status, @Nullable Integer done, @Nullable Integer total);
+    void onFailure(Exception cause);
+  }
+
+  private interface MultiDownloadListener {
+    void onSuccess();
+    void onFailure(Exception cause);
+  }
+
   public BundleDownloader(OkHttpClient client) {
     mClient = client;
   }
 
   public void downloadBundleFromURL(
       final DevBundleDownloadListener callback,
       final File outputFile,
+      final File bundlesContainerFile,
       final String bundleURL,
       final @Nullable BundleInfo bundleInfo,
-      final BundleDeltaClient.ClientType clientType) {
+      final BundleDeltaClient.ClientType clientType,
+      final CreateBundleURLLambda bundleURLCreator) {
     downloadBundleFromURL(
-        callback, outputFile, bundleURL, bundleInfo, clientType, new Request.Builder());
+      callback,
+      outputFile,
+      bundlesContainerFile,
+      bundleURL,
+      bundleInfo,
+      clientType,
+      new Request.Builder(),
+      bundleURLCreator
+    );
   }
 
   public void downloadBundleFromURL(
       final DevBundleDownloadListener callback,
       final File outputFile,
+      final File bundlesContainerFile,
       final String bundleURL,
       final @Nullable BundleInfo bundleInfo,
       final BundleDeltaClient.ClientType clientType,
-      Request.Builder requestBuilder) {
+      Request.Builder requestBuilder,
+      final CreateBundleURLLambda bundleURLCreator) {
+
+    mDevBundlesContainer = new DevBundlesContainer(bundleURL);
+    // DOWNLOAD INITIAL BUNDLE
+    downloadBundle(new DownloadListener() {
```

Comment:
can the method here be renamed to downloadInitialBundle so that the comment can be removed?